### PR TITLE
WX-927 GCPBATCH retry with more memory and error message fixes, Centaur tests

### PR DESCRIPTION
**Pull request by mcovarr**
### Description

The PR exercises the "retry with more memory" Centaur tests on the GCP Batch backend. Minimal changes to production code, all of which are in the GCP Batch backend:

- The constant`RunnableUtils#MountPoint` was created with value `/mnt/disks/cromwell_root` and applied where appropriate.
- A copy/paste bug in code brought over from PAPIv2 was corrected (the `/cromwell_root` of PAPIv2 has become `/mnt/disks/cromwell_root` in Batch), using the constant described above.
- If a job fails, the *last* event message is now propagated rather than the first event message. The first event message is often a benign state transition, while the last event message is more likely to contain the actual reason for job failure.
 
Unfortunately Cromwell does not allow for dynamic backend selection (i.e. the backend name cannot be a variable), which necessitated copy/paste/renaming the Centaur test WDLs from their PAPIv2 versions, hence the magnitude of these diffs.

The existing `preemptible_and_memory_retry ` Centaur test is heavily tailored to the quirks of Papi v2: a preemptible PAPI VM deletes itself and depends on the Lifesciences system mistaking that for a preemption event. tbh this is kind of a weird test and as I don't know how to induce a preemption on demand, I simply `ignore`d the GCPBATCH version.

#### `CHANGELOG.md`
 - [x] I updated `CHANGELOG.md` in this PR
 - [ ] I assert that this change shouldn't be included in `CHANGELOG.md` because it doesn't impact community users

#### Terra Release Notes
 - [ ] I added a suggested release notes entry in this Jira ticket
 - [x] I assert that this change doesn't need Jira release notes because it doesn't impact Terra users

### CODE REVIEW

File: supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/runnable/RunnableCommands.scala [hunk 1]
```diff
@@ -163,7 +163,7 @@ object RunnableCommands {
 
   def checkIfStderrContainsRetryKeys(retryLookupKeys: List[String]): String = {
     val lookupKeysAsString = retryLookupKeys.mkString("|")
-    s"grep -E -q '$lookupKeysAsString' /cromwell_root/stderr ; echo $$? > /cromwell_root/memory_retry_rc"
+    s"grep -E -q '$lookupKeysAsString' /mnt/disks/cromwell_root/stderr ; echo $$? > /mnt/disks/cromwell_root/memory_retry_rc"
```
Comment (AlexITC) [inline supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/runnable/RunnableCommands.scala:166]: We should consider placing "/mnt/disks/cromwell_root" as a constant and reuse it when needed.

**Review by aednichols — APPROVED** [review]
Nice, today I learned about the `backend: "GCPBATCH"` attribute.

File: centaur/src/main/resources/standardTestCases/gcpbatch_preemptible_and_memory_retry.test [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: gcpbatch_preemptible_and_memory_retry
+testFormat: workflowfailure
+# The original version of this test seems to have been tailored to the quirks of Papi v2 in depending on the misdiagnosis of its own VM deletion as a preemption event. GCP Batch perhaps more correctly diagnoses the VM deletion as a weird non-preemption happening, but that frustrates the logic of this test.
+# Disabling this as it's not possible to induce a real preemption.
```
Comment (jgainerdewar) [inline centaur/src/main/resources/standardTestCases/gcpbatch_preemptible_and_memory_retry.test:4]: Thank you for adding this comment!

**Review by jgainerdewar — APPROVED** [review]
Nice! I believe this closes WX-1618.

**Pull request merged**